tesseract_collision/bullet/src/bullet_utils.cpp

```diff
@@ -530,7 +530,7 @@ const tesseract_common::VectorIsometry3d& CollisionObjectWrapper::getCollisionGe
 void CollisionObjectWrapper::getAABB(btVector3& aabb_min, btVector3& aabb_max) const
 {
   getCollisionShape()->getAabb(getWorldTransform(), aabb_min, aabb_max);
-  const btScalar& d = getContactProcessingThreshold();
+  const btScalar& d = getContactProcessingThreshold() / 2.0;
```

Comment:
I think it might be better to make this calculation here, this way we do not need to worry about if bullet uses shape processing threshold any other place.

``` c++
void BulletCastBVHManager::onCollisionMarginDataChanged()
{
  auto margin = static_cast<btScalar>(contact_test_data_.collision_margin_data.getMaxCollisionMargin() / 2.0);
```

Reply:
Eh, I commented with some explanation, but that apparently got lost. Will retry tomorrow.

Reply:
@rjoomen Friendly ping

Reply:
The reason to not do it in onCollisionMarginDataChanged() is that the ContactProcessingThreshold of the object is then set to half the margin. This threshold is also used [here](https://github.com/bulletphysics/bullet3/blob/d1a4256b3a019117f2bb6cb8c63d6367aaf512e2/src/BulletCollision/CollisionDispatch/btCollisionDispatcher.cpp#L77) or [here](https://github.com/tesseract-robotics/tesseract/blob/de5f4d8e4cd3a5fa9737fc5571727f02b8b69016/tesseract_collision/bullet/src/bullet_cast_simple_manager.cpp#L410), where I do not believe a halved margin would be expected. By only halving the margin in getAABB() we make sure it's only used in the broadphase.

Reply:
@Levi-Armstrong I've expanded this PR, see the modified description.

Reply:
> The reason to not do it in onCollisionMarginDataChanged() is that the ContactProcessingThreshold of the object is then set to half the margin. This threshold is also used [here](https://github.com/bulletphysics/bullet3/blob/d1a4256b3a019117f2bb6cb8c63d6367aaf512e2/src/BulletCollision/CollisionDispatch/btCollisionDispatcher.cpp#L77) or [here](https://github.com/tesseract-robotics/tesseract/blob/de5f4d8e4cd3a5fa9737fc5571727f02b8b69016/tesseract_collision/bullet/src/bullet_cast_simple_manager.cpp#L410), where I do not believe a halved margin would be expected. By only halving the margin in getAABB() we make sure it's only used in the broadphase.

This makes sense, it would be good to document what you just explained where you make the change.